### DISCUSSION TITLE
Best practice for deploying seaweedfs gateway without storing data locally, only metadata

### DISCUSSION THREAD

**Discussion by lenlacko** [seaweedfs/seaweedfs#4927] [Q&A] [answered]
Hi,

I have been running seaweedfs on my k8s cluster for a while with multiple replicas of master, volume, filer, and s3 inside filer. There is a gateway running on the leader master pod pointing to a remote storage predefined and mounted using weed shell. The PVs are stored on a local disk using local path provisioner, also the disk is going to run out of storage soon. To prevent this i tried using remote.uncache as suggested in the weed shell docs, but i haven't seen any effect of it. My goal is to have the properties of the data, to stay connected to my remote storage and also free some space on my disk, so the actual data is deleted from the local path. What is the best practice to achieve this?

Thanks in advance

**Comment by chrislusf** [comment, marked as answer]
"i haven't seen any effect of it" 

Did you see output such as `"Uncache %+v ... "`?

After `remote.uncache`, you may need to compact the volumes to see the effect.

**Reply by lenlacko** [reply]
Thanks for the quick reply!
I think it will solve my problem.
Also thanks for Seaweedfs, it is a great tool!

**Comment by kmlebedev** [comment]
@lenlacko Hello, it seems your case will be interesting to other users. It would be great if you could briefly describe deploying seaweedfs gateway on the [wiki](https://github.com/seaweedfs/seaweedfs/wiki)(Misc Use Case Examples) and highlight the advantages of this solution.
